src/lib/containers/FacetsPlotNav.tsx

```diff
@@ -0,0 +1,296 @@
+  //     const plotPointData: any = event.points[0]
+  //     const domain = plotPointData.data.domain
+  //     const newMarkerLineWidth: number = isHover ? 1 : 0
+  //     this.state.plotData[(domain.row * CHARTS_PER_ROW) + domain.column].marker.line.width[plotPointData.pointNumber] = newMarkerLineWidth
+  //     this.setState({datarevision: this.state.datarevision + 1})
+  //   }
+  // }
+
+  /**
+   * Handle click event
+   */
+  public handleClick = (event: any) => {
+    if (event.points && event.points[0]) {
+      const plotPointData: any = event.points[0]
+      const facetName = plotPointData.data.name
+      const facetValueClicked = plotPointData.data.facetEnumerationValues[plotPointData.pointNumber]
+      // update the facet and selected index
+      const chartSelectionIndex = plotPointData.pointNumber
+      const { isAllFilterSelectedForFacet = {} } = this.props
+      isAllFilterSelectedForFacet[facetName] = false
+      const lastFacetSelection = {
+        selector: SELECT_SINGLE_FACET,
+        facetValue: facetValueClicked,
+        columnName: facetName,
+      } as FacetSelection
+      this.props.updateParentState!({
+        chartSelectionIndex,
+        lastFacetSelection,
+        isAllFilterSelectedForFacet,
+      })
+
+      // run the query with the selected facet value
+      const queryRequest: QueryBundleRequest = this.props.getLastQueryRequest!()
+      const { selectedFacets = [] } = queryRequest.query
+      const specificFacet = selectedFacets!.find(el => el.columnName === facetName)!
+      if (!specificFacet) {
+        const facetColumnValuesRequest: FacetColumnValuesRequest = {
+          facetValues: [facetValueClicked],
+          concreteType:
+            'org.sagebionetworks.repo.model.table.FacetColumnValuesRequest',
+          columnName: facetName,
+        }
+        selectedFacets.push(facetColumnValuesRequest)
+        // align the reference to selectedFacets
+        queryRequest.query.selectedFacets = selectedFacets
+      } else {
+        specificFacet.facetValues = [facetValueClicked]
+      }
+
+      queryRequest.query.offset = 0
+      this.props.executeQueryRequest!(queryRequest)
+    }
+    this.setState({
+      isResetPossible: true
+    })
+  }
+
+  public render() {
+    const {
+      data,
+      isLoadingNewData,
+      loadingScreen,
+      // rgbIndex,
+      asyncJobStatus,
+    } = this.props
+    // while loading
+    if (isLoadingNewData || !data) {
+      return (
+        <div className="SRC-loadingContainer SRC-centerContentColumn">
+          {/*
+            check loading screen is not undefined or null and show
+            it if so
+          */}
+          {!!loadingScreen && loadingScreen}
+          <div>{asyncJobStatus && asyncJobStatus.progressMessage}</div>
+        </div>
+      )
+    }
+    const plotData = this.extractPropsData(data!)
+    const showMoreButton = data!.facets!.length > CHARTS_PER_ROW && (
```

Comment:
show more button should not be visible if item count is less than max charts in a row.